### PR TITLE
Add 1D property plotter

### PR DESCRIPTION
**Pull request by dstansby**
This adds a way to visualse the currently selected track/property using a 1D plot made with Matplotlib. See below for a static example - the plot automatically updates when a different track ID or property is selected in napari. @KristinaUlicna perhpas you could give this a go and post any feedback you have below? There are probably a few layout/display things that could be improved, but I thought I'd open early to get some feedback.

Screenshot 2022-05-23 at 15 07 05

### CODE REVIEW

**Comment by KristinaUlicna** [comment]
This looks fantastic, I'm gonna have a look at more files right now!

**Comment by KristinaUlicna** [comment]
Thanks for fixing the `napari-matplotlib` error with the extra requirement! Here's a little preview of what the plugin outputs - I think it looks great!

Screenshot 2022-05-23 at 16 15 00

**Comment by quantumjot** [comment]
So cool - love it!

**Comment by KristinaUlicna** [comment]
### Some ideas for improvements:

- [ ] [Tree viz] Y-axis + label i.e. 'Time [frames]'
- [ ] [Tree viz] Title - I usually call the trees by their root cell ID i.e. 'Lineage Tree #67'
- [ ] [Tree viz] It would be great to know which cell I'm currently looking at when traversing the tree. How about a horizontal line crossing the tree at the position which is displayed in the middle panel?
- [ ] [Tree viz] If we could draw a white box (or something like this) around the cell in the middle panel so we can monitor which cell we initially clicked on & have highlighted on the tree, that'd be gold. Just thinking about ideas how to not lose track (pun intended) of which cell we're visualising.

**Comment by dstansby** [comment]
Thanks for the great feedback! I'll try and address the property graph ideas in this PR - for the others there's either open issues/PRs, or I've just opened a couple of issues as they can all be tackled independently. I've updated your comment (https://github.com/lowe-lab-ucl/arboretum/pull/60#issuecomment-1134838731) with the issues or PRs for each of the suggestions.

**Comment by dstansby** [comment]
I think this is ready for review now. @KristinaUlicna since this is pretty complicated already in terms of code changes, if/when this is merged I'll open your other suggestions (all of which I think are great ideas that should be implemented!) as individual issues.

File: napari_arboretum/util.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        """
+        Optional method that derived classes can override to do something
+        when the tracks layer changes.
+        """
+
+    def on_track_id_change(self) -> None:
+        """
+        Optional method that derived classes can override to do something
+        when the track ID is changed.
+        """
```
Comment (paddyroddy) [inline napari_arboretum/util.py:47]: Should these not have `raise NotImplementedError`?

File: napari_arboretum/_tests/test_plugin.py [hunk 2]
```diff
@@ -40,7 +41,7 @@ def test_colormap_change(viewer_plugin):
     """
     viewer, plugin = viewer_plugin
     id = 140
```
Comment (paddyroddy) [inline napari_arboretum/_tests/test_plugin.py:43]: Could this `id` not be set as a variable at the top of the file, seeing as it is being used more than once?

File: napari_arboretum/visualisation/base_plotter.py [hunk 3]
```diff
@@ -134,3 +126,85 @@ def get_qwidget(self) -> QWidget:
...
+        """
+        Draw track ID. Where this is drawn is up to the implmenation, and could
+        e.g. be the plot title or plot legend.
+        """
+
+    def redraw(self) -> None:
+        """
+        Optional method that can be implemented by sub-classes to re-draw
+        a figure after the plot and labels have been set.
+        """
```
Comment (paddyroddy) [inline napari_arboretum/visualisation/base_plotter.py:210]: Again these should have `raise NotImplementedError`

File: setup.cfg [hunk 4]
```diff
@@ -11,3 +11,7 @@ max-complexity = 18
 exclude =
   __init__.py
   napari_arboretum/_tests
+
+[isort]
+profile = black
+line_length = 79
```
Comment (paddyroddy) [inline setup.cfg:17]: Currently, this code base is using `black`'s (sensible) default of `88`. Doing this here will mean that `isort` is using `black`'s profile with line length `79` when the rest of the code will be using `88`. IMO, we shouldn't be changing unless there's a good reason. The ability to change to something `!= 88` is for legacy code bases where `black` has been introduced at a much later date, which is not the case here

**Review by paddyroddy — CHANGES_REQUESTED** [review]
Main concern is line length bit for `isort`

File: napari_arboretum/util.py (hunk 1 above)
Comment (dstansby) [reply napari_arboretum/util.py:47]: No, it's optional to override them, and if they're not overriden the intent is to do nothing.

File: setup.cfg (hunk 4 above)
Comment (dstansby) [reply setup.cfg:17]: I think I just put this in as a default, and I'm not that bothered about the particlar value. Should I just change this line to 88?

Comment (dstansby) [reply setup.cfg]: I changed this to 88

File: napari_arboretum/util.py (hunk 1 above)
Comment (paddyroddy) [reply napari_arboretum/util.py:47]: How about `pass` then? I find functions without anything in them uncomfortable personally. Looks like they're unfinished

File: napari_arboretum/visualisation/base_plotter.py (hunk 3 above)
Comment (paddyroddy) [reply napari_arboretum/visualisation/base_plotter.py]: `pass`?

File: setup.cfg (hunk 4 above)
Comment (paddyroddy) [reply setup.cfg]: You don't need to. `isort`'s `black` profile will use `black` defaults which is 88

**Review by paddyroddy — APPROVED** [review]
I'm happy, but think we should remove the line length bit in `setup.cfg` unless you feel strongly about it.

The other thing is empty function definitions, I've had a brief look but can't find anything in style guide about it. Partly because not sure what to Google, empty function definition doesn't seem to work...

**Pull request merged**